exifread/classes.py

```diff
@@ -248,9 +246,11 @@ def _process_tag(self, ifd, ifd_name: str, tag_entry, entry, tag: int, tag_name,
 
         field_offset = offset
         values = None
-        if field_type == 2:
+        if field_type == 2:  # ascii strings
             values = self._process_field2(ifd_name, tag_name, count, offset)
-        else:
+        elif field_type == 7:  # undefined
+            values = self._process_field7(ifd_name, tag_name, count, offset)
```

Comment:
It's better not to handle data format 7 (undefined) type in function `_process_field`, which is used to parse signed/unsigned numbers.

Reply:
That's why I moved this special case into another function `_process_field7`.